src/lib/routes/admin-api/context.ts

```diff
@@ -8,54 +8,158 @@ import {
     CREATE_CONTEXT_FIELD,
     UPDATE_CONTEXT_FIELD,
     DELETE_CONTEXT_FIELD,
+    NONE,
 } from '../../types/permissions';
 import { IUnleashConfig } from '../../types/option';
 import { IUnleashServices } from '../../types/services';
 import ContextService from '../../services/context-service';
 import { Logger } from '../../logger';
 import { IAuthRequest } from '../unleash-types';
 
-class ContextController extends Controller {
-    private logger: Logger;
+import { OpenApiService } from '../../services/openapi-service';
+import { ContextSchema } from '../../openapi/spec/context-schema';
+import { createResponseSchema } from '../../openapi';
+import { serializeDates } from 'lib/types/serialize-dates';
+
+interface ContextParam {
+    contextField: string;
+}
 
+export class ContextController extends Controller {
     private contextService: ContextService;
 
+    private openApiService: OpenApiService;
+
+    private logger: Logger;
+
     constructor(
         config: IUnleashConfig,
-        { contextService }: Pick<IUnleashServices, 'contextService'>,
+        {
+            contextService,
+            openApiService,
+        }: Pick<IUnleashServices, 'contextService' | 'openApiService'>,
     ) {
         super(config);
+        this.openApiService = openApiService;
         this.logger = config.getLogger('/admin-api/context.ts');
         this.contextService = contextService;
 
-        this.get('/', this.getContextFields);
-        this.post('/', this.createContextField, CREATE_CONTEXT_FIELD);
-        this.get('/:contextField', this.getContextField);
-        this.put(
-            '/:contextField',
-            this.updateContextField,
-            UPDATE_CONTEXT_FIELD,
-        );
-        this.delete(
-            '/:contextField',
-            this.deleteContextField,
-            DELETE_CONTEXT_FIELD,
-        );
-        this.post('/validate', this.validate, UPDATE_CONTEXT_FIELD);
+        this.route({
+            method: 'get',
+            path: '',
+            handler: this.getContextFields,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'getContextFields',
+                    responses: {
+                        200: createResponseSchema('contextSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: '',
+            handler: this.createContextField,
+            permission: CREATE_CONTEXT_FIELD,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'createContextField',
+                    responses: {
```

Comment:
We should also add request schemas, for runtime request body validation. 

See for example `updateFeatureSchema`/`UpdateFeatureSchema` and `createRequestSchema`.